libraries/botbuilder-core/src/activityHandlerBase.ts

```diff
@@ -149,6 +153,10 @@ export class ActivityHandlerBase {
         return;
     }
 
+    protected async onInvokeActivity(context: TurnContext): Promise<void|any> {
```

Comment:
why do you need any in the return type?

Reply:
We try to not use any

Reply:
InvokeResponse is defined in botbuilder and activityHandler files live in botbuilder-core. I wil need to move InvokeResponse to botbuilder-core to be able to use it explicitly here instead of any, which seems like a bit of refactoring (moving invokeResponse to botbuilder-core would mean that the interfaces related to requests and responses are now split between 2 modeules). IN the interest of avoiding that complexity, i am marking the return as any

---
In reply to: [383515276](https://github.com/microsoft/botbuilder-js/pull/1738#discussion_r383515276) [](ancestors = 383515276)

Reply:
The plan is to bring the code from TeamsActivityHandler in here. The pattern used there avoids the explicit use of InvokeResponse because it maps the request-reply to a conventional typed function call. The result of that function call being the body of the InvokeRequest. Basically we hide the "send an InvokeResponseActivity" behavior that has confused everyone whos looked at it behind a conventional function call model. Consolidating the code with TeamsActivityHandler is not part of this PR. Please open a bug and assign to me when you merge.